lib/carto/ghost_tables_manager.rb

```diff
@@ -134,50 +135,23 @@ def sync
             regenerated_tables.any?{|t| t.name == dropped_table.name}
         end
 
+      return regenerated_tables, renamed_tables, new_tables, dropped_tables
+    end
+
+    def sync
+      regenerated_tables, renamed_tables, new_tables, dropped_tables = fetch_altered_tables
+
       # Update table_id on UserTables with physical tables with changed oid. Should go first.
-      regenerated_tables.each(&:regenerate_user_table)
+      regenerated_tables.each(&:regenerate_user_table) if regenerated_tables
```

Comment:
If `regenerated_tables` is an empty array, it will do nothing.
```suggestion
      regenerated_tables.each(&:regenerate_user_table)
```